modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -538,3 +538,101 @@ RETURN [stop IN nodes(p) | stop.name] AS r1,
 
 2+d|Rows: 1
 |===
+
+
+== Planning shortest paths queries
+
+This section describes the operators used when planning shortest path queries.
+For readers not familiar with Cypher execution plans and operators, it is recommended to first read the section xref:planning-and-tuning/execution-plans.adoc[].
+
+There are two operators used to plan `SHORTEST` queries:
+
+* xref:planning-and-tuning/operators/operators-detail.adoc#query-plan-stateful-shortest-path-all[`StatefulShortestPath(All)`] - uses a unidirectional breadth-first search algorithm to find shortest paths from a previously matched start node to an end node that has not yet been matched.
+
+* xref:planning-and-tuning/operators/operators-detail.adoc#query-plan-stateful-shortest-path-into[`StatefulShortestPath(Into)`] - uses a bidirectional breadth-first search (BFS) algorithm, where two simultaneous BFS invocations are performed, one from the left boundary node and one from the right boundary node.
+
+
+`StatefulShortestPath(Into)` is used by the planner when both boundary nodes in the shortest path have been matched in a `MATCH` clause previous to that containing the `SHORTEST` keyword.
```

Comment:
The wording makes this seem to be a plan that is decided at runtime, rather than based on cardinality estimates. For example, on the UK railway dataset I get only one node matching each end node pattern, but it is still planned as `StatefulShortestPath(All)`:
```
PROFILE
MATCH (n:Station {name: "Denmark Hill"}), (m:Station WHERE m.name CONTAINS "Vic")
MATCH p = SHORTEST 1 (n)((l)-[e:LINK]-(r) WHERE e.distance < 10){,5}(m)
RETURN p
```

Reply:
Good point. Let me look into this